File: pbt-java/src/main/java/pl/edu/bogdan/pbt/Calculator.java
package pl.edu.bogdan.pbt;

public class Calculator {
    private final int value;

    public Calculator(int value) {
        this.value = value;
    }

    public Calculator add(int other) {
        return new Calculator(value + other);
    }

    public Calculator sub(int other) {
        return new Calculator(value - other);
    }

    public Calculator mul(int other) {
        return new Calculator(value * other);
    }

    public Calculator div(int other) {
        return new Calculator(value / other);
    }

    public int getCurrentValue() {
        return value;
    }
}

File: pbt-scala/src/main/scala/pl/edu/bogdan/pbt/Calculator.scala
package pl.edu.bogdan.pbt

class Calculator(value: Int) {

  def add(other: Int): Calculator = new Calculator(value + other)

  def sub(other: Int): Calculator = new Calculator(value - other)

  def mul(other: Int): Calculator = new Calculator(value * other)

  def div(other: Int): Calculator = new Calculator(value / other)

  def getCurrentValue: Int = value

}
